Sources/MessagingPush/MessagingPush.swift

```diff
@@ -1,14 +1,37 @@
 import CioTracking
 import Foundation
+#if canImport(UserNotifications) && canImport(UIKit)
+import UIKit
+import UserNotifications
+#endif
 
-public protocol MessagingPushInstance: AutoMockable {
+public protocol MessagingPushInstance {
     func registerDeviceToken(_ deviceToken: String)
     func deleteDeviceToken()
     func trackMetric(
         deliveryID: String,
         event: Metric,
         deviceToken: String
     )
+
+    #if canImport(UserNotifications)
+    @discardableResult
+    // sourcery:Name=didReceiveNotificationRequest
+    // sourcery:IfCanImport=UserNotifications
+    func didReceive(
+        _ request: UNNotificationRequest,
+        withContentHandler contentHandler: @escaping (UNNotificationContent) -> Void
+    ) -> Bool
+
+    // sourcery:IfCanImport=UserNotifications
+    func serviceExtensionTimeWillExpire()
+
+    func userNotificationCenter(
+        _ center: UNUserNotificationCenter,
+        didReceive response: UNNotificationResponse,
+        withCompletionHandler completionHandler: @escaping () -> Void
+    ) -> Bool
+    #endif
```

Comment:
this is for rich push. originally, this code was only in the APN module but FCM needs access to these methods, too.